src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -52,6 +53,29 @@ public static DirectoryInfo CreateDirectory(string path)
         public static DirectoryInfo CreateDirectory(string path, UnixFileMode unixCreateMode)
             => CreateDirectoryCore(path, unixCreateMode);
 
+        /// <summary>
+        /// Creates a uniquely-named, empty directory in the current user's temporary directory.
+        /// </summary>
+        /// <param name="prefix">An optional string to add to the beginning of the subdirectory name.</param>
+        /// <returns>An object that represents the directory that was created.</returns>
+        /// <exception cref="ArgumentException"><paramref name="prefix" /> contains a directory separator.</exception>
+        /// <exception cref="IOException">A new directory cannot be created.</exception>
+        public static unsafe DirectoryInfo CreateTempSubdirectory(string? prefix = null)
+        {
+            EnsureNoDirectorySeparators(prefix);
+
+            string path = CreateTempSubdirectoryCore(prefix);
+            return new DirectoryInfo(path, isNormalized: true);
+        }
+
+        private static void EnsureNoDirectorySeparators(string? value, [CallerArgumentExpression("value")] string? paramName = null)
+        {
+            if (!string.IsNullOrEmpty(value) && value.IndexOfAny(s_directorySeparators) != -1)
```

Comment:
```suggestion
            if (value.AsSpan().IndexOfAny(DirectorySeparators) < 0)
```
and change DirectorySeparators to constant.

Reply:
The issue is that on Windows there are 2 and on Unix there is only 1. 
To make it a constant, won’t I need to ifdef this code? What type would you expect the constant to be?

Reply:
It could be a string.